tests/FSharp.Compiler.ComponentTests/FSharpChecker/FindReferences.fs

```diff
@@ -301,3 +301,104 @@ let x = MyType()
             "FileProgram.fs", 6, 8, 14
         ])
     }
+
+
+module ActivePatterns =
+
+    /// https://github.com/dotnet/fsharp/issues/14206
+    [<Fact>]
+    let ``Finding references to an active pattern case shouldn't find other cases`` () =
+        let source = """
+let (|Even|Odd|) v =
+    if v % 2 = 0 then Even else Odd
+match 2 with
+| Even -> ()
+| Odd -> ()
+"""
+        let fileName, options, checker = singleFileChecker source
+
+        let symbolUse = getSymbolUse fileName source "Even" options checker |> Async.RunSynchronously
+
+        checker.FindBackgroundReferencesInFile(fileName, options, symbolUse.Symbol, fastCheck = true)
+        |> Async.RunSynchronously
+        |> expectToFind [
+            fileName, 2, 6, 10
+            fileName, 3, 22, 26
+            fileName, 5, 2, 6
+        ]
+
+    [<Fact>]
+    let ``We don't find references to cases from other active patterns with the same name`` () =
+
+        let source = """
+module One =
+
+    let (|Even|Odd|) v =
+        if v % 2 = 0 then Even else Odd
+    match 2 with
+    | Even -> ()
+    | Odd -> ()
+
+module Two =
+
+    let (|Even|Steven|) v =
+        if v % 3 = 0 then Steven else Even
+    match 2 with
+    | Even -> ()
+    | Steven -> ()
+"""
+
+        let fileName, options, checker = singleFileChecker source
+
+        let symbolUse = getSymbolUse fileName source "Even" options checker |> Async.RunSynchronously
+
+        checker.FindBackgroundReferencesInFile(fileName, options, symbolUse.Symbol, fastCheck = true)
+        |> Async.RunSynchronously
+        |> expectToFind [
+            fileName, 4, 10, 14
+            fileName, 5, 26, 30
+            fileName, 7, 6, 10
+        ]
+
+    [<Fact>]
+    let ``We don't find references to cases the same active pattern defined in a different file`` () =
+
+        let source = """
+let (|Even|Odd|) v =
+    if v % 2 = 0 then Even else Odd
+match 2 with
+| Even -> ()
+| Odd -> ()
+"""
+        SyntheticProject.Create(
+            { sourceFile "First" [] with Source = source },
+            { sourceFile "Second" [] with Source = source }
+        ).Workflow {
+            placeCursor "First" "Even"
+            findAllReferences (expectToFind [
+                "FileFirst.fs", 3, 6, 10
+                "FileFirst.fs", 4, 22, 26
+                "FileFirst.fs", 6, 2, 6
+            ])
+        }
+
+    [<Fact>]
+    let ``We find active patterns in other files when there are signature files`` () =
+
+        SyntheticProject.Create(
+            { sourceFile "First" [] with
+                Source = "let (|Even|Odd|) v = if v % 2 = 0 then Even else Odd"
+                SignatureFile = AutoGenerated },
```

Comment:
So do I understand correctly that renaming an active pattern case will still not lead to its renaming in the signature file (and hence there'll be a compiler error)?

Reply:
Yes. Signature files ruin everything 😭 

To make this work would probably require significant changes.

Reply:
Yeah. Created https://github.com/dotnet/fsharp/issues/14969 and added a test here documenting the broken behavior.